torch/_higher_order_ops/flex_attention.py

```diff
@@ -781,15 +781,19 @@ def sdpa_dense_backward(
     from torch._dynamo._trace_wrapped_higher_order_op import TransformGetItemToIndex
 
     # Get outputs before calling repeat interleave
-    actual_grad_query = torch.empty_like(query)
-    actual_grad_key = torch.empty_like(key)
-    actual_grad_value = torch.empty_like(value)
+    actual_grad_query = torch.empty_like(query, memory_format=torch.preserve_format)
+    actual_grad_key = torch.empty_like(key, memory_format=torch.preserve_format)
```

Comment:
Okay I did more tinkering and actually this should be empty_strided.

empty_like only works for the non-overlapping case but doesn't help w/ the sliced case.
Or even better we figure out the right inductor lowering to mimix empty like: https://github.com/pytorch/pytorch/blob/5f0901e57341eb9865102c1caa3d986a0c4ae3bd/torch/_inductor/kernel/flex_attention.py#L2320-L2344

However we need to have a larger tensor .. okay actualy the code I linked above just doesn't make sense I will re-thinking this tomorrow. But suffice it to say I still the the main problem is that our meta func is wrong

Reply:
@drisspg the default for empty_like is preserve_format, did you change these for clarity?